### ISSUE TITLE
Problems with astropy version 1.0.5

### ISSUE THREAD

**Issue by wschoenell** [astropy/ccdproc#241]
I don't know if I should post here or on astropy/astropy, but here it goes...
If I run `python -c "import ccdproc; ccdproc.test()"` with the latest astropy version (1.0.5), I end up with a LOT of errors like the error below. It seems to be caused by this commit: astropy/astropy@018e7f23208355fc39453ec05b29a00b8863c56f

	ccd_data = CCDData([[-1.0856306 ,  0.99734545,  0.2829785 , ..., -1.36347154,
			   ....15776154, -0.88186588, ..., -0.47226641,
			   0.58196437,  0.97061286]])
	tmpdir = local('/private/var/folders/r_/r_8gdqz175x0hkk012ndjplm0000gn/T/pytest-0/test_wcs_attribute0/pytest-0/test_write_unit_to_hdu0')

		def test_write_unit_to_hdu(ccd_data, tmpdir):
			ccd_unit = ccd_data.unit
	>       hdulist = ccd_data.to_hdu()

	/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/tests/test_ccddata.py:112: 
	_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
	/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/ccddata.py:208: in to_hdu
		dummy_ccd = CCDData(data=[1], meta=fits.Header(), unit="adu")
	/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/ccddata.py:105: in __init__
		super(CCDData, self).__init__(*args, **kwd)
	_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

	self = CCDData([1]), arg = (), kwd = {'data': [1], 'meta': , 'unit': 'adu'}, flags = None

		def __init__(self, *arg, **kwd):
			# Remove the flag argument from input.
			flags = kwd.pop('flags', None)
	
			# Initialize with the parent...
			super(NDDataArray, self).__init__(*arg, **kwd)
	
			# ...then reset uncertainty to force it to go through the
			# setter logic below. In base NDData all that is done is to
			# set self._uncertainty to whatever uncertainty is passed in.
			self.uncertainty = self._uncertainty
	
			# Same thing for mask.
			self.mask = self._mask
	
			# Initial flags because it is no longer handled in NDData
			# or NDDataBase.
	>       data = arg[0]  # do this after parent __init__, to validate the args
	E       IndexError: tuple index out of range

**Comment by mwcraig** [comment]
@wschoenell -- once again, thanks! The issue belongs here. As far as I can tell the problem the `data=` in the first argument of https://github.com/astropy/ccdproc/blob/master/ccdproc/tests/pytest_fixtures.py#L42

The `NDData` class clearly indicates the first argument is required, so shouldn't be a keyword, but that never mattered until until astropy/astropy#4130 was merged in, which assumes (correctly) that the first argument will be present.

If you are still interested in fixing #240 you could also fix this. Otherwise I'll do a fix (or have a student do a fix) Monday or Tuesday.

**Comment by wschoenell** [comment]
Thanks for you reply. I will moving to another continent this week so I don't think that I'll have time to look on this... So, please go ahead. If you still need help I can re-join this issue in 2 weeks.

**Comment by crawfordsm** [comment]
Just trying this out with some other code and it looks like the astropy update also breaks the multiply function in ccdproc--so not sure if it something so simple as fixing the testing suite.    I'll delve into it a little further right now.

**Comment by wschoenell** [comment]
Indeed. We found this problem after updating ccdproc and having the error below on script that was working before.

```python
   ccd = ccdproc.flat_correct(ccd, current_master_flat)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/log_meta.py", line 68, in wrapper
    result = func(*args, **kwd)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/core.py", line 481, in flat_correct
    use_flat.unit)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/ccddata.py", line 270, in multiply
    scale_uncertainty=True)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/ccddata.py", line 262, in _ccddata_arithmetic
    meta=self.meta)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ccdproc/ccddata.py", line 105, in __init__
    super(CCDData, self).__init__(*args, **kwd)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/nddata/compat.py", line 101, in __init__
    data = arg[0]  # do this after parent __init__, to validate the args
IndexError: tuple index out of range

**Comment by crawfordsm** [comment]
Okay this should now be fixed in the just merged #242    Let me know if you continue to have any problems with this.

**Comment by wschoenell** [comment]
Thanks guys!
